src/cow-react/modules/swap/pure/Row/RowDeadline/index.tsx

```diff
@@ -35,27 +35,48 @@ export function getNonNativeOrderDeadlineTooltip() {
 export interface RowDeadlineProps extends Omit<RowSlippageProps, 'allowedSlippage'> {
   toggleSettings: () => void
   isEthFlow: boolean
+  isExpertMode: boolean
+  isWrapOrUnwrap: boolean
   symbols?: (string | undefined)[]
   displayDeadline: string
   styleProps?: RowStyleProps
   userDeadline: number
 }
 
 export function RowDeadlineContent(props: RowDeadlineProps) {
-  const { userDeadline, showSettingOnClick, toggleSettings, displayDeadline, isEthFlow, symbols, styleProps } = props
+  const {
+    showSettingOnClick,
+    toggleSettings,
+    displayDeadline,
+    isEthFlow,
+    isExpertMode,
+    isWrapOrUnwrap,
+    symbols,
+    styleProps,
+  } = props
 
-  if (!isEthFlow || userDeadline > MINIMUM_ETH_FLOW_DEADLINE_SECONDS) return null
+  if ((!isEthFlow && !isExpertMode) || isWrapOrUnwrap) {
+    return null
+  }
 
   return (
     <StyledRowBetween {...styleProps}>
       <RowFixed>
         <TextWrapper>
-          <Trans>
-            Transaction expiration{' '}
-            <ThemedText.Warn display="inline-block" override>
-              (modified)
-            </ThemedText.Warn>
-          </Trans>
+          {isEthFlow ? (
+            <Trans>
+              Transaction expiration{' '}
+              <ThemedText.Warn display="inline-block" override>
+                (modified)
+              </ThemedText.Warn>
+            </Trans>
+          ) : showSettingOnClick ? (
+            <ClickableText onClick={toggleSettings}>
+              <Trans>Transaction expiration</Trans>
+            </ClickableText>
+          ) : (
+            <Trans>Transaction expiration</Trans>
+          )}
```

Comment:
i don't love 2 ternary operators in a row, but just an ubernit here

Reply:
I just copied what's done for https://github.com/cowprotocol/cowswap/blob/7e4b9e2daa5e8178307a64782dc6082899a01491/src/cow-react/modules/swap/pure/Row/RowSlippageContent/index.tsx#L60-L73

But I'll update both